Repository/ProfileRepository.php

```diff
@@ -0,0 +1,359 @@
+<?php
+
+namespace App\Repository;
+
+use App\Api\Structure\FigureParams;
+use App\Entity\Profile\Profile;
+use App\Entity\Profile\ProfileHasLanguage;
+use App\Entity\Profile\ProfileHasService;
+use App\Entity\Profile\ProfileOpeningHours;
+use App\Entity\User;
+use App\Model\Google\ViewPort;
+use App\Utils\Strings;
+use Doctrine\ORM\Query\Expr\Join;
+use Doctrine\ORM\QueryBuilder;
+
+class ProfileRepository
+{
+
+    /**
+     * @return Profile[]
+     */
+    public function findCompanions(
+        Profile $currentProfile,
+        ?float $lat,
+        ?float $lng,
+        ?int $limit,
+        ?int $offset,
+        ?array $sex,
+        ?int $ageFrom,
+        ?int $ageTo,
+        ?FigureParams $figureParams,
+        ?array $ethnicity,
+        ?array $orientation,
+        ?array $smoker,
+        ?array $serviceIds,
+        ?array $languageIds,
+        ?bool $isPornStar,
+        ?bool $escort,
+        ?bool $escortToSociety,
+        ?bool $isIndependent,
+        ?bool $traveling,
+        ?bool $hasVideo,
+        ?ViewPort $viewPort,
+        string $orderBy = self::ORDER_BY_DISTANCE,
+        ?bool $active = null,
+        ?bool $followed = null,
+        ?Profile $assignedToProfile = null,
+        ?bool $approved = null,
+        ?bool $bookable = null
+    ) {
+        $queryBuilder = $this->findQueryBuilder(
+            $currentProfile,
+            Profile::TYPE_COMPANION,
+            $limit,
+            $offset,
+            null,
+            null,
+            $sex,
+            $ageFrom,
+            $ageTo,
+            $figureParams,
+            $ethnicity,
+            $orientation,
+            $smoker,
+            $serviceIds,
+            $languageIds,
+            $isPornStar,
+            $escort,
+            $escortToSociety,
+            null,
+            null,
+            null,
+            false,
```

Comment:
U něčeho takového bych již dnes použil named parametry.
On PHPStorm nám sice ukáže kam se to posílá, ale když to příjde někomu na review tak nemá tušení co tenhle false vlastně znamená. Kdyby se použil named parametr tak je to ihned v code review jasné